### PR TITLE
OADP-4454: Fixing configuration error in Azure installation docs

### PR DESCRIPTION
**Pull request by anarnold97**
### JIRA

* [OADP-4454](https://issues.redhat.com//browse/OADP-4454)

I have reworked the logic for OADP to remove the Azure CLI commands
 
### Versions

* OCP 4.12 → branch/enterprise-4.12
* OCP 4.13 → branch/enterprise-4.13
* OCP 4.14 → branch/enterprise-4.14
* OCP 4.15 → branch/enterprise-4.15
* OCP 4.16 → branch/enterprise-4.16

#### OADP preview:

* [Configuring Microsoft Azure](https://78670--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-azure.html#migration-configuring-azure_installing-oadp-azure)

##### MTC docs impacted:

* [MTC - Configuring Microsoft Azure](https://78670--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-azure.html#migration-configuring-azure_installing-oadp-azure)
* [Installing the Migration Toolkit for Containers - Configuring Microsoft Azure](https://78670--ocpdocs-pr.netlify.app/openshift-enterprise/latest/migration_toolkit_for_containers/installing-mtc.html#migration-configuring-azure_installing-mtc)

### QE review:
- [X] QE has approved this change. Amos Mastbaum approved ([LGTM](https://issues.redhat.com/browse/OADP-4454) in JIRA)

Additional information:

### CODE REVIEW

File: modules/migration-configuring-azure.adoc [hunk 1]
```diff
@@ -92,12 +92,31 @@ $ az storage container create \
...
-  AZURE_CLIENT_SECRET=`az ad sp create-for-rbac --name "velero" \
-  --role "Contributor" --query 'password' -o tsv` \
-  AZURE_CLIENT_ID=`az ad sp list --display-name "velero" \
-  --query '[0].appId' -o tsv`
+$ AZURE_SUBSCRIPTION_ID=`az account list --query '[?isDefault].id' -o tsv`
+----
++
+[source,terminal]
+----
+$ AZURE_TENANT_ID=`az account list --query '[?isDefault].tenantId' -o tsv`
```
Comment (anarnold97) [inline modules/migration-configuring-azure.adoc:100]:
```suggestion
  AZURE_TENANT_ID=`az account list --query '[?isDefault].tenantId' -o tsv`
```

**Comment by anarnold97** [comment]
> @anarnold97 seems like section 11 and 14 are duplicated i think you only wnat to leave 14

@amastbau - which is really weird as I only added section 9 and 10! LOL

**Comment by anarnold97** [comment]
@amastbau - i think it is caused by a missing `ifdef`.

So if we change lines 117 to 134 and include an `ifndef` not to include `installing-oadp-azure` variable, which should pair with the `ifdef` to include `installing-oadp-azure` variable.

Please can you check again.

File: modules/migration-configuring-azure.adoc [hunk 2]
```diff
@@ -92,14 +108,30 @@ $ az storage container create \
...
 ----
 
+. Create a service principal with the `Contributor` role, assigning a specific `--role` and `--scopes`:
++
+[source,terminal]
+----
+$ AZURE_CLIENT_SECRET=`az ad sp create-for-rbac --name "velero" \
+                                                --role "Contributor" \
+                                                --query 'password' -o tsv \
+                                                --scopes /subscriptions/<subscription_ID>/<resource_groups>/<resource_group_group_name>`
```
Comment (anarnold97) [inline modules/migration-configuring-azure.adoc:122]:
```suggestion
                                                --scopes /subscriptions/$subscriptionID/resourceGroups/$resourceGroup`
```

File: modules/migration-configuring-azure.adoc [hunk 3]
```diff
@@ -6,29 +6,45 @@
...
 ** The source and target clusters must be in the same region.
 ** The source and target clusters must have the same storage class.
 ** The storage class must be compatible with snapshots.
 endif::[]
+ifdef::installing-oadp-azure[]
+
+Tools that use {azure-short} services should always have restricted permissions to make sure that {azure-short} resources are safe. Therefore, instead of having applications sign in as a fully privileged user, {azure-short} offers service principals. An {azure-short} service principal is a name that can be used with applications, hosted services, or automated tools.
+
+This identity is used for access to resources.
+
```
Comment (anarnold97) [inline modules/migration-configuring-azure.adoc:33]:
```suggestion
```

File: modules/migration-configuring-azure.adoc [hunk 4]
```diff
@@ -92,14 +107,30 @@ $ az storage container create \
...
+----
+
+. Create a service principal with the `Contributor` role, assigning a specific `--role` and `--scopes`:
++
+[source,terminal]
+----
+$ AZURE_CLIENT_SECRET=`az ad sp create-for-rbac --name "velero" \
+                                                --role "Contributor" \
+                                                --query 'password' -o tsv \
+                                                --scopes /subscriptions/$subscriptionID/resourceGroups/$resourceGroup`
```
Comment (mail2nadeem92) [inline modules/migration-configuring-azure.adoc:121]: A small suggestion, As we already fetching the Azure subscriptionID in line 110 in the variable AZURE_SUBSCRIPTION_ID and the azure resourceGroup using variable AZURE_RESOURCE_GROUP , So we can use them directly here.  So please update them here.

--scopes /subscriptions/$AZURE_SUBSCRIPTION_ID/resourceGroups/$AZURE_RESOURCE_GROUP

Comment (anarnold97) [inline modules/migration-configuring-azure.adoc:121]:
```suggestion
                                                --scopes /subscriptions/$AZURE_SUBSCRIPTION_ID/resourceGroups/$AZURE_RESOURCE_GROUP`
```

File: modules/migration-configuring-azure.adoc [hunk 5]
```diff
@@ -92,14 +107,30 @@ $ az storage container create \
...
+                                                --scopes /subscriptions/$AZURE_SUBSCRIPTION_ID/resourceGroups/$AZURE_RESOURCE_GROUP`
 ----
++
+The CLI generates a password for you. Ensure you capture the password.
 
+. After creating the service principal, obtain the client id.
++
+[source,terminal]
+----
+$ AZURE_CLIENT_ID=`az ad sp list --display-name "velero" --query '[0].appId' -o tsv`
```
Comment (anarnold97) [inline modules/migration-configuring-azure.adoc:130]:
```
$ AZURE_CLIENT_ID=`az ad app credential list --id <your_app_id>`
```

Comment (anarnold97) [inline modules/migration-configuring-azure.adoc:130]:
```suggestion
$ AZURE_CLIENT_ID=`az ad app credential list --id <your_app_id>`
```

File: modules/migration-configuring-azure.adoc [hunk 6]
```diff
@@ -92,14 +107,30 @@ $ az storage container create \
...
++
+The CLI generates a password for you. Ensure you capture the password.
 
+. After creating the service principal, obtain the client id.
++
+[source,terminal]
+----
+$ AZURE_CLIENT_ID=`az ad app credential list --id <your_app_id>`
+----
+endif::[]
```
Comment (anarnold97) [inline modules/migration-configuring-azure.adoc:132]:
```suggestion
+
[NOTE]
====
For this to be successful, you must know your {azure-short} application ID.
====
endif::[]
```

File: modules/migration-configuring-azure.adoc [hunk 7]
```diff
@@ -92,14 +107,35 @@ $ az storage container create \
...
+                                                --scopes /subscriptions/$AZURE_SUBSCRIPTION_ID/resourceGroups/$AZURE_RESOURCE_GROUP`
+----
++
+The CLI generates a password for you. Ensure you capture the password.
+
+. After creating the service principal, obtain the client id.
++
+[source,terminal]
+----
+$ AZURE_CLIENT_ID=`az ad app credential list --id <your_app_id>`
```
Comment (anarnold97) [inline modules/migration-configuring-azure.adoc:130]: @mail2nadeem92 - are you happy with this?
Thanks

File: modules/migration-configuring-azure.adoc [hunk 8]
```diff
@@ -6,29 +6,44 @@
 
 :_mod-docs-content-type: PROCEDURE
 [id="migration-configuring-azure_{context}"]
-= Configuring Microsoft Azure
+= Configuring Microsoft {azure-full}
```
Comment (sheriff-rh) [inline modules/migration-configuring-azure.adoc:9]: The build preview says Microsoft Microsoft Azure.

File: modules/migration-configuring-azure.adoc [hunk 9]
```diff
@@ -6,29 +6,44 @@
...
 
 ifdef::installing-3-4,installing-mtc[]
-You configure a Microsoft Azure Blob storage container as a replication repository for the {mtc-full} ({mtc-short}).
+You configure a {azure-full} Blob storage container as a replication repository for the {mtc-full} ({mtc-short}).
 endif::[]
 ifdef::installing-oadp-azure[]
-You configure Microsoft Azure for the OpenShift API for Data Protection (OADP).
+You configure Microsoft {azure-full}Azure for {oadp-first}.
```
Comment (sheriff-rh) [inline modules/migration-configuring-azure.adoc:15]: And this says Microsoft Microsoft AzureAzure.

**Review by sheriff-rh — COMMENTED** [review]
Some small corrections suggested. Please squash your commits prior to merge review. Thank you!

**Review by sheriff-rh — COMMENTED** [review]
Some small corrections suggested. Please squash your commits prior to merge review. Thank you!

File: modules/migration-configuring-azure.adoc (hunk 8 above)
Comment (anarnold97) [inline modules/migration-configuring-azure.adoc:9]:
```suggestion
= Configuring {azure-full}
```

File: modules/migration-configuring-azure.adoc [hunk 10]
```diff
@@ -6,29 +6,44 @@
...
-= Configuring Microsoft Azure
+= Configuring {azure-full}
 
 ifdef::installing-3-4,installing-mtc[]
-You configure a Microsoft Azure Blob storage container as a replication repository for the {mtc-full} ({mtc-short}).
+You configure a {azure-full} Blob storage container as a replication repository for the {mtc-full} ({mtc-short}).
 endif::[]
 ifdef::installing-oadp-azure[]
-You configure Microsoft Azure for the OpenShift API for Data Protection (OADP).
+You configure Microsoft {azure-full}Azure for {oadp-first}.
```
Comment (anarnold97) [inline modules/migration-configuring-azure.adoc:15]:
```suggestion
You configure {azure-full} for {oadp-first}.
```

**Comment by anarnold97** [comment]
squashed ready for merge review

File: modules/migration-configuring-azure.adoc [hunk 11]
```diff
@@ -92,14 +107,35 @@ $ az storage container create \
...
+----
+$ AZURE_CLIENT_SECRET=`az ad sp create-for-rbac --name "velero" \
+                                                --role "Contributor" \
+                                                --query 'password' -o tsv \
+                                                --scopes /subscriptions/$AZURE_SUBSCRIPTION_ID/resourceGroups/$AZURE_RESOURCE_GROUP`
+----
++
+The CLI generates a password for you. Ensure you capture the password.
+
+. After creating the service principal, obtain the client id.
```
Comment (stevsmit) [inline modules/migration-configuring-azure.adoc:126]:
```suggestion
. After creating the service principal, obtain the client id:
```

**Review by stevsmit — COMMENTED** [review]
A few smaller nits, like preceding all ocmmands with ". . . by entering the following command:", but nothing that should hold up a review. Good work.

**Pull request merged**